### DISCUSSION TITLE
Why do we add 'reverse edges' in hetero graph?

### DISCUSSION THREAD

**Discussion by Jeriousman** [pyg-team/pytorch_geometric#3976] [Q&A] [answered]
I wish to know this in a detailed way. I do not know how it helps for the model as even tho it is a reverse edge, it is still a different edge. For example, 

['user', 'watched', 'movie'] and ['movie'. 'watched-by', 'user'] are very similar but still different edges. How does that help to improve for some tasks?

**Comment by rusty1s** [comment, marked as answer]
Reverse edge types are necessary to allow that every node types receives new information during message passing. For example, in a bipartite graph with `user` and `movie` types, and a single relation `watched`, message passing does only send information from users to movies, not the other way around. In that case, only movie embeddings are learned, and you would miss out on the information of users that watched similar movies.

**Reply by Jeriousman** [reply]
Thank you again for the detailed information
